src/config.js

```diff
@@ -43,7 +44,7 @@ module.exports = (store) => {
      * @param {string} [key] - the config key to get
      * @param {Object} [options] - options
      * @param {AbortSignal} [options.signal] - abort this config read
-     * @returns {Promise<unknown>}
+     * @returns {Promise<Config | any>}
```

Comment:
Can't we get `any` out of here?

Reply:
If you omit the `key` you get `Config`, otherwise you get `any` because it could be a string or an array or and object, etc.  We could make `key` mandatory but they we'd just lose `Config` as a return type.  Might be worth doing anyway TBF.

Reply:
I've made `key` mandatory, since we don't treat it as optional in js-ipfs.